tools/vm-migrator/migrator/instance.py

```diff
@@ -0,0 +1,397 @@
+    try:
+        waited_time = rate_limit.wait()  # wait before starting the task
+        logging.info(f"  task: waited for {waited_time} secs")
+        compute = get_compute()
+        logging.info("Starting Instance %s ", (instance_name))
+        result = compute.instances().start(project=project,
+                                           zone=zone,
+                                           instance=instance_name).execute()
+        wait_for_zonal_operation(compute, project, zone, result['name'])
+        return instance_name
+    except Exception as ex:
+        logging.error(ex)
+        print(ex)
+        raise ex
+
+
+def delete_instance(compute, project, zone, name):
+    logging.info("Deleting Instance %s ", (name))
+    return compute.instances().delete(project=project, zone=zone,
+                                      instance=name).execute()
+
+
+def wait_for_zonal_operation(compute, project, zone, operation):
+    logging.info('Waiting for operation to finish...')
+    while True:
+        result = compute.zoneOperations().get(project=project,
+                                              zone=zone,
+                                              operation=operation).execute()
+
+        if result['status'] == 'DONE':
+            logging.info("done.")
+            if 'error' in result:
+                raise Exception(result['error'])
+            return result
+
+        time.sleep(10)
+
+
+def wait_for_regional_operation(compute, project, region, operation):
+    logging.info('Waiting for operation to finish...')
+    while True:
+        result = compute.regionOperations().get(project=project,
+                                                region=region,
+                                                operation=operation).execute()
+
+        if result['status'] == 'DONE':
+            logging.info("done.")
+            if 'error' in result:
+                raise Exception(result['error'])
+            return result
+
+        time.sleep(10)
+
+
+def delete(project, zone, name):
+    try:
+        waited_time = rate_limit.wait()  # wait before starting the task
+        logging.info(f"  task: waited for {waited_time} secs")
+        compute = get_compute()
+        image = machine_image.get(project, name)
+        if (image):
+            logging.info("Found machine image can safely delete the instance")
+            delete_operation = delete_instance(compute, project, zone, name)
+            wait_for_zonal_operation(compute, project, zone,
+                                     delete_operation['name'])
+            return name
+        else:
+            raise Exception(
+                "Cant Delete the instance as machine image not found")
+    except Exception as ex:
+        logging.error(ex)
+        print(ex)
+        raise ex
+
+
+def get_region_from_zone(zone):
+    match = re.search(r"(\w+)-(\w+)-(\w+)", zone)
+    if (len(match.groups()) != 3):
+        raise Exception('Invalid Zone Format')
```

Comment:
What kind of exception is this? If you raise a custom exception, it would be possible to handle that type of exception isolated from other types of exceptions. If you raise a generic `Exception`, it cannot be handled isolatedly.